### PR TITLE
Fixes avsc library's types to not use unsafe function/eval executions

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Modifies all references of `new Function` in types to use regular function/callbacks
- Fixes webpack build to support the new implementation

**TODO:**
- Right now unit tests for types pass fine but there are 6 other tests that fail. Since we don't use protocols or read/write streams I am not spending more time on it right now.
- Have changed the webpack node environment so that we can exclude the injection of some node libraries that have unsafe `eval` or `new Function` references.
- So far tested the build in CDAP UI and it seems to work fine. Will fix the failing unit tests for protocols and streams in a subsequent PR.

### CODE REVIEW

File: lib/types.js [hunk 1]
```diff
@@ -1851,35 +1850,47 @@ RecordType.prototype._getConstructorName = function () {
...
     }
   }
-  var outerBody = 'return function ' + this._getConstructorName() + '(';
-  outerBody += innerArgs.join() + ') {\n' + innerBody + '};';
-  var Record = new Function(outerArgs.join(), outerBody).apply(undefined, ds);
 
+  var self = this;
+  function ConstructorFunction(oArgs) {
+    var constructorName = self._getConstructorName();
+    const stringifyName = (name) => `'${name}'`;
```
Comment (elfenheart) [inline lib/types.js]: not being used?

**Review by elfenheart — COMMENTED** [review]
I think you'll have to update the package.json version as well so that we can change the version when publishing to npm

**Comment by ajainarayanan** [comment]
@elfenheart Yes planned on bumping the version once the PR looked ok. Have added the change.

File: lib/types.js (hunk 1 above)
Comment (ajainarayanan) [reply lib/types.js]: Removed.

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Pull request merged**